lib/Target/TfheRust/TfheRustEmitter.cpp

```diff
@@ -778,24 +787,56 @@ FailureOr<std::string> TfheRustEmitter::convertType(Type type) {
         return (type.isUnsigned() ? std::string("u") : "") + "i" +
                std::to_string(width.value());
       })
-      // TODO(#474): Generalize to any encrypted uint.
-      .Case<EncryptedUInt3Type>(
-          [&](auto type) { return std::string("Ciphertext"); })
+      .Case<EncryptedUInt2Type>(mark_ciphertext)
```

Comment:
I believe you can put all the types into a comma-separated template argument list, see for example https://github.com/llvm/llvm-project/blob/4f77677c3b383da99e46fdb29e7b8de63eb72447/mlir/lib/Dialect/LLVMIR/IR/LLVMTypes.cpp#L835-L848

Similarly for below.